DistributionUpdate/PackageUpdate/PackageInfoTools.g

```diff
@@ -336,23 +336,34 @@ UpdatePackageInfoFiles := function(pkgdir)
                 "\n  This is not allowed, so the info file will not be changed\n");
           has_error := true;
         fi;
-        if IsBound(infon.Date) then
-          date := infon.Date;
-          if not IsString(date) and Length(date) = 10 and date{[3,6]} = "//" and
-            ForAll( date{ [1,2,4,5,7,8,9,10] }, IsDigitChar ) then
-            Print("  ERROR (", info.PackageName, "): the date ", date,
-                  "should be a string of the form `dd/mm/yyyy'\n" );
-            has_error := true;
-          else
-            date := List( SplitString( date, "/" ), Int);
-            if not date[1] in [1..31] and
-                   date[2] in [1..12] and
-                   date[3] > 1999 and # GAP 4 appeared in 1999
-                   date[1] > DaysInMonth( date[2], date[3] ) then
+        if IsBound(info.Date) then
+          if IsString(infon.date) and Length(infon.Date) = 10 and infon.Date{[3,6]} = "//" and
+              ForAll( infon.Date{ [1,2,4,5,7,8,9,10] }, IsDigitChar ) then
+            date := List( SplitString( infon.Date, "/" ), Int); #here date=[dd,mm,yyyy]
+            #the format dd/mm/yyyy is ambigous and can be confused with mm/dd/yyyy
+            #if it is clear from the date that the format is mm/dd/yyyy print a message
+            if date[2] in [13..31] and date[1] in [1..12] then
               Print("  ERROR (", info.PackageName, "): the date ", date,
-                    " is invalid\n" );
+                    "should be a string of the form `yyyy-mm-dd` or `dd/mm/yyyy'\n" );
               has_error := true;
             fi;
+          elif IsString(infon.Date) and Length(infon.Date) = 10 and infon.Date{[5,8]} = "--" and
+              ForAll( infon.Date{ [1,2,3,4,6,7,9,10] }, IsDigitChar ) then
+            date := List( SplitString( infon.Date, "-" ), Int);
+            date := Permuted(date, (1,3)); #sort such that date=[dd,mm,yyyy]
+          else
+            Print("  ERROR (", info.PackageName, "): the date ", date,
+                    "should be a string of the form `yyyy-mm-dd` or `dd/mm/yyyy'\n" );
+              has_error := true;
+          fi;
+          if IsBound(date) and (not
```

Comment:
Perhaps like this would be better, to avoid warning twice?
```suggestion
          if not has_error and IsBound(date) and (not
```
Then again, `has_error` might have been set earlier... so alternatively, you could `Unbind(date);` above, after the errors are reported.

Reply:
I use the alternative method with Unbind now.